### ISSUE TITLE
Dropping to Script Editor is broken if scrolling past end of file is enabled

### ISSUE THREAD

**Issue by KoBeWi** [godotengine/godot#59822]
### Godot version

3.5 beta3

### System information

Windows 10 x64

### Issue description

When you enable `text_editor/cursor/scroll_past_end_of_file` in editor settings and try to drop a file into the script, dragging over the area under the script will cause scrolling down and makes the editor unusable:
![godot_RN52PSq21w](https://user-images.githubusercontent.com/2223172/161407741-ca58acff-6c36-4d8f-82a3-a5da8e6ddd5b.gif)

Does not happen on master.

### Steps to reproduce

1. Enable scroll past end of file
2. Open some script that can scroll
3. Try dropping some file

**Comment by akien-mga** [comment]
I can confirm the regression in 3.5 beta 2, 3.5 beta 3 and latest `3.x`.
It doesn't happen in 3.4.4-stable or 3.5 beta 2, so the regression happened between beta 2 and beta 3.

**Comment by akien-mga** [comment]
I haven't tested but from reviewing the changelog I have a hunch that it might be because of #55355. CC @ConteZero

**Comment by ConteZero** [comment]
Yes, the problem is in the following code:

```
		if (drag_action && can_drop_data(mm->get_position(), get_viewport()->gui_get_drag_data())) {
			drag_caret_force_displayed = true;
			Point2 mp = get_local_mouse_position();
			int row, col;
			_get_mouse_pos(Point2i(mp.x, mp.y), row, col);
			cursor_set_line(row, true);
			cursor_set_column(col);
			if (row <= get_first_visible_line()) {
				_scroll_lines_up();
			} else if (row >= get_last_full_visible_line()) {
				_scroll_lines_down();
			}
			dragging_selection = true;
			update();
		}
```
But I don't know how to fix it.

**Comment by KoBeWi** [comment]
The easiest "fix" would be removing this part:
```C++
if (row <= get_first_visible_line()) {
	_scroll_lines_up();
} else if (row >= get_last_full_visible_line()) {
	_scroll_lines_down();
}
```
It would disable scrolling when dragging at the edges, but it's better than breaking the editor. From my experience, most dragging happens on a single screen.

**Comment by akien-mga** [comment]
Fixed by #61980.
